code/modules/mob/living/carbon/human/life/handle_regular_hud_updates.dm

```diff
@@ -99,6 +99,24 @@
 		// Rewrite idea : divide life() into organs (Eyes...) and have flags in the roles if they overwrite the functions of those organs.
 		// Basically, the problem here is that abstract things (HUD icons) are handled as the same time as "organs" things (seeing in the dark.)
 		var/datum/role/vampire/V = isvampire(src)
+		if (V)
+			var/i = 1
+			for (var/image/I in V.cached_images)
+				I.loc = null
+				src.client.images -= I
+			for (var/mob/living/carbon/C in view(7,src))
+				var/obj/item/weapon/nullrod/N = locate(/obj/item/weapon/nullrod) in get_contents_in_object(C)
+				if (N)
+					if (i > V.cached_images.len)
+						var/image/I = image('icons/mob/mob.dmi', loc = C, icon_state = "vampnullrod")
+						I.plane = VAMP_ANTAG_HUD_PLANE
+						V.cached_images += I
+						src.client.images += I
+					else
+						V.cached_images[i].loc = C
+						src.client.images += V.cached_images[i]
+					i++
+
```

Comment:
maybe remove all entries with `pos=null`, it seems `V.cached_images` will get cluttered easily if the nullrod gets passed between people. I dont know how gc works, does it fire just based on `loc=null`?

Reply:
The point is not to delete the image, but to hide it. A reference is kept in the role. Images must be "hard" deleted, unlike datums, so I felt caching it was better than deleting it.

Reply:
good point